### PR TITLE
Refactors team antagonists and roundend report handling

### PR DESCRIPTION
**Pull request by JohnFulpWillard**
## About The Pull Request

Refactors teams and subtypes
* Makes use of ``as anything`` and ``\improper`` (like I already did for Antagonists in https://github.com/tgstation/tgstation/pull/63469)
* Removes many single letter variables
* Moves team code to the team file
* Makes team objectives only allow objectives for SpacemanDMM
* Makes members and objectives clear when a team is deleted
* Generalizes team's add_objective
* Checks for show_roundend_report in roundend
* Replaces ``get_team == src`` with a simple ``if(mind in members)``
* Makes Bloodbrothers fit using parent's endround report.
* Removes the unused procs ``is_solo`` and ``antag_listing_footer``
* Makes replaces ``get_team_antag`` ``specific`` with ``include_subtypes`` to be consistent with Antag helpers (and because I thought it looked better that way)

image
image

This is what it looks like in-game currently (2 bloodbrother teams, 1 solo abductor, 2 traitors- 1 traitor was also a bloodbrother):
image

## Why It's Good For The Game

I looked at the code for this and it make me legitimately sad, a lot of it was meant to be placeholders, so hopefully this will be better long-term

## Changelog

:cl:
refactor: Teams used by Antagonists have been refactored, please report any bugs you might find.
spellcheck: Team antagonists are now improper.
spellcheck: Bloodbrothers now show their team antagonist name in admin panels.
/:cl:

### CODE REVIEW

File: code/__HELPERS/roundend.dm [hunk 1]
```diff
@@ -589,19 +589,25 @@
 	var/list/all_teams = list()
 	var/list/all_antagonists = list()
 
-	for(var/datum/team/A in GLOB.antagonist_teams)
-		all_teams |= A
+	for(var/datum/team/teams as anything in GLOB.antagonist_teams)
```
Comment (SuperNovaa41) [inline code/__HELPERS/roundend.dm:592]:
```suggestion
	for(var/datum/team/team as anything in GLOB.antagonist_teams)
```
it's singular NOT plural i hate youi hate youi hate youi hate you

File: code/modules/antagonists/_common/antag_team.dm [hunk 2]
```diff
@@ -60,3 +69,36 @@ GLOBAL_LIST_EMPTY(antagonist_teams)
...
+			continue
+
+		if(include_subtypes && istype(antagonists, antag_datum))
+			list += antagonists
+		else if(antagonists.type == antag_datum)
+			list += antagonists
+
+	return list
+
+//Builds section for the team
```
Comment (SuperNovaa41) [inline code/modules/antagonists/_common/antag_team.dm:86]:
```suggestion
/// Builds section for the team
```

File: code/modules/antagonists/_common/antag_team.dm [hunk 3]
```diff
@@ -60,3 +69,36 @@ GLOBAL_LIST_EMPTY(antagonist_teams)
...
+/datum/team/proc/antag_listing_entry()
+	//NukeOps:
+	// Jim (Status) FLW PM TP
+	// Joe (Status) FLW PM TP
+	//Disk:
+	// Deep Space FLW
+	var/list/parts = list()
+	parts += "<b>[antag_listing_name()]</b><br>"
+	parts += "<table cellspacing=5>"
+	for(var/datum/antagonist/A in get_team_antags())
```
Comment (SuperNovaa41) [inline code/modules/antagonists/_common/antag_team.dm:96]: var

**Review by SuperNovaa41 — CHANGES_REQUESTED** [review]

File: code/modules/antagonists/_common/antag_team.dm [hunk 4]
```diff
@@ -60,3 +69,36 @@ GLOBAL_LIST_EMPTY(antagonist_teams)
 
 
 	return "<div class='panel redborder'>[report.Join("<br>")]</div>"
+
+/datum/team/proc/get_team_antags(antag_datum, include_subtypes = TRUE)
+	var/datum/list = list()
```
Comment (SuperNovaa41) [inline code/modules/antagonists/_common/antag_team.dm:74]: im confused why don't you just do 
```
var/list/antag_list = list()
```
or something why `var/datum/list`?

**Review by SuperNovaa41 — CHANGES_REQUESTED** [review]

File: code/modules/antagonists/_common/antag_team.dm (hunk 4 above)
Comment (JohnFulpWillard) [reply code/modules/antagonists/_common/antag_team.dm:74]: because i was sleep deprived, fixed it now.

**Pull request merged**